src/backend/distributed/utils/citus_stat_tenants.c

```diff
@@ -236,26 +257,51 @@ AttributeTask(char *tenantId, int colocationId, CmdType commandType)
 
 /*
  * AnnotateQuery annotates the query with tenant attributes.
+ * if the query has a partition key, we annotate it with the partition key value and colocationId
+ * if the query doesn't have a partition key and if it's a schema-based tenant, we annotate it with the colocationId only.
  */
 char *
 AnnotateQuery(char *queryString, Const *partitionKeyValue, int colocationId)
 {
-	if (StatTenantsTrack == STAT_TENANTS_TRACK_NONE || partitionKeyValue == NULL)
+	if (StatTenantsTrack == STAT_TENANTS_TRACK_NONE || colocationId ==
+		INVALID_COLOCATION_ID)
```

Comment:
nit: let's put this in a more reasonable place (it's one of the things citus_indent messes up sometimes).
```suggestion
	if (StatTenantsTrack == STAT_TENANTS_TRACK_NONE || 
	    colocationId == INVALID_COLOCATION_ID)
```